prow/flagutil/github.go

```diff
@@ -112,9 +126,16 @@ func (o *GitHubOptions) GitHubClient(secretAgent *secret.Agent, dryRun bool) (cl
 	return o.GitHubClientWithLogFields(secretAgent, dryRun, logrus.Fields{})
 }
 
+// GitHubClientWithAccessToken creates a GitHub client from an access token.
+func (o *GitHubOptions) GitHubClientWithAccessToken(token string) github.Client {
+	return github.NewClient(func() []byte { return []byte(token) }, func(content []byte) []byte {
```

Comment:
That's not how the censor works -- this censoring function will replace every full message in the log with `"CENSORED"` regardless of what the log message was.

Reply:
If we can DRY this out with the `CensoringFormatter` that would be good

Reply:
> That's not how the censor works -- this censoring function will replace every full message in the log with "CENSORED" regardless of what the log message was.

I know how it works, but thanks for sharing anyway. And do you think this is wrong? (Sorry not getting the point).

> If we can DRY this out with the CensoringFormatter that would be good

I don't see it obvious to use CensoringFormatter here as it requests a proper logrus Formatter and secrets list, but I do agree with your DRY point, this also repeats with `secret.Agent`, may I ask for further suggestion on how do you want to change this? Thanks!

Reply:
The `func([]byte) []byte` is meant to replace sensitive parts of the incoming message with censored placeholders. If you always simply replace the entirety of the content with the string `"CENSORED"` you get no logging at all, every log line will just be `"CENSORED"`. Is that what you want?

Reply:
Oh, now I see the point. Thanks.